x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -283,9 +332,21 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
           .join(' AND ');
       }
 
-      const url = await discoverLocator.getRedirectUrl({
+      // For multi-metric or population jobs, we add the selected entity for links to
+      // Log Rate Analysis, so they can be restored as part of the search filter.
+      if (withWindowParameters && props.anomaly.entityName && props.anomaly.entityValue) {
+        if (kqlQuery !== '') {
+          kqlQuery += ' AND ';
+        }
+
+        kqlQuery = `"${escapeQuotes(props.anomaly.entityName)}":"${escapeQuotes(
+          props.anomaly.entityValue + ''
+        )}"`;
+      }
+
+      return {
```

Comment:
As discussed, currently you have to hit the browser back button twice to return to the Anomaly Explorer / SMV from the log rate analysis page.

Looks like the first redirect adds (empty) `filters` to the URL:

```
filters:!(),searchQuery:(match_all:())
```

Is there anything we can do to prevent the extra redirect?

Reply:
I was able to fix it in [6e6f148](https://github.com/elastic/kibana/pull/175289/commits/6e6f148b5ff8a805ce2a3be9da5096d0484a67c8). The problem was that the window parameters being passed on from Anomaly Detection needed to be snapped to the bars of the log rate analysis histogram chart which caused an URL state update. I changed the behaviour so updates to the baseline+deviation time range won't cause a push to the browser's history.

Reply:
I extended the functional tests to include a check that when navigating back just once the Single Metric Viewer/Anomaly Explorer loads correctly again. [d0b2222](https://github.com/elastic/kibana/pull/175289/commits/d0b2222bef2ba88712b50ffecf7601ed686410e0)